unittests/bootseq_tests.cpp

```diff
@@ -370,4 +369,101 @@ BOOST_FIXTURE_TEST_CASE( bootseq_test, bootseq_tester ) {
     } FC_LOG_AND_RETHROW()
 }
 
+BOOST_FIXTURE_TEST_CASE( stake_lock_test, bootseq_tester ) {
+    try{
+        // Create eosio.msig and eosio.token
+        create_accounts({N(eosio.msig), N(eosio.token), N(eosio.ram), N(eosio.ramfee), N(eosio.stake), N(eosio.vpay), N(eosio.bpay), N(eosio.saving) });
+
+        set_code_abi(N(eosio.msig),
+                     contracts::eosio_msig_wasm(),
+                     contracts::eosio_msig_abi().data());//, &eosio_active_pk);
+        set_code_abi(N(eosio.token),
+                     contracts::eosio_token_wasm(),
+                     contracts::eosio_token_abi().data()); //, &eosio_active_pk);
+
+        // Set privileged for eosio.msig and eosio.token
+        set_privileged(N(eosio.msig));
+        set_privileged(N(eosio.token));
+
+         // Create SYS tokens in eosio.token, set its manager as eosio
+        auto max_supply = core_from_string("100.0000"); 
+        auto initial_supply = core_from_string("100.0000");
+        create_currency(N(eosio.token), config::system_account_name, max_supply);
+        // Issue the genesis supply of 1 billion SYS tokenbs to eosio.system
+        issue(N(eosio.token), config::system_account_name, config::system_account_name, initial_supply);
+
+         // Create genesis accounts
+        for( const auto& a : test_genesis ) {
+            create_account( a.aname, config::system_account_name );
+        }
+
+        deploy_contract();
+
+        // Buy ram and stake cpu and net for each genesis accounts
+        for( const auto& a : test_genesis ) {
+            auto ib = a.initial_balance;
+            auto ram = 1000;
+            auto net = (ib - ram) / 2;
+            auto cpu = ib - net - ram;
+
+            auto r = buyram(config::system_account_name, a.aname, asset(ram));
+            BOOST_REQUIRE( !r->except_ptr );
+
+            r = delegate_bandwidth(N(eosio.stake), a.aname, asset(net), asset(cpu));
+            BOOST_REQUIRE( !r->except_ptr );
+        }
+
+        auto producer_candidates = {N(proda)};
+
+        register_producer( N(proda));
+
+        // Vote for producers
+        auto votepro = [&]( account_name voter, vector<account_name> producers ) {
+            std::sort( producers.begin(), producers.end() );
+            base_tester::push_action(config::system_account_name, N(voteproducer), voter, mvo()
+                    ("voter",  name(voter))
+                    ("proxy", name(0) )
+                    ("producers", producers)
+            );
+        };
+
+        votepro( N(b1), { N(proda)} );
+        votepro( N(whale2), {N(proda)} );
+        votepro( N(whale3), {N(proda)} );
+        votepro( N(whale4), {N(proda)} );
+
+        // No producers will be set, since the total activated stake is less than 150,000,000
+        produce_blocks_for_n_rounds(2); // 2 rounds since new producer schedule is set when the first block of next round is irreversible
```

Comment:
@AndriiAndrusyshyn not sure but seems that we can safely delete those lines
```
        // Since the total vote stake is more than 150,000,000, the new producer set will be set
        produce_blocks_for_n_rounds(2); // 2 rounds since new producer schedule is set when the first block of next round is irreversible
        active_schedule = control->head_block_state()->active_schedule;
```